### PR TITLE
Add Boiler Efficiency output for Boiler objects

### LINKED ISSUE #8064: Boiler objects are missing an output variable for efficiency
Issue overview
--------------
Boiler objects are missing an output variable for efficiency. Specifically the Boiler:HotWater object does not have the variable. From the RDD, and consistent with the docs, here are the output variables:

HVAC,Average,Boiler Heating Rate [W]
HVAC,Sum,Boiler Heating Energy [J]
HVAC,Average,Boiler Gas Rate [W]
HVAC,Sum,Boiler Gas Energy [J]
HVAC,Average,Boiler Inlet Temperature [C]
HVAC,Average,Boiler Outlet Temperature [C]
HVAC,Average,Boiler Mass Flow Rate [kg/s]
HVAC,Average,Boiler Ancillary Electric Power [W]
HVAC,Sum,Boiler Ancillary Electric Energy [J]
HVAC,Average,Boiler Part Load Ratio []

Boiler:HotWater has a potentially non-constant efficiency because it has an efficiency curve. Currently you would have to divide the output variables for Boiler Heating Rate / Boiler Gas Rate as a post-processing step.  The chiller objects have a comparable generic Chiller COP output variable.

Boiler:Steam should also be checked, but it might only have a constant efficiency value.

### Details
Some additional details for this issue (if relevant):
 - EnergyPlus V9.3

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by matthew-larson**
Pull request overview
---------------------
 - Fixes #8064 
 - This pull request adds output variables for Boiler (Steam) Efficiency for the Boiler:HotWater and Boiler:Steam objects.  This provides an efficiency output variable for the Boiler objects similar to other plant objects like the Chiller objects providing a Chiller COP output variable.  For simplification purposes, I changed the variable Effic for boiler efficiency at design conditions to NomEffic, similar to NomCap, and created a new variable BoilerEff to be the outputted variable for each timestep.  Unit tests were created to ensure the BoilerEff variable result is correct.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts

### CODE REVIEW

File: doc/input-output-reference/src/overview/group-plant-equipment.tex [hunk 1]
```diff
@@ -4159,6 +4161,10 @@ \subsubsection{Outputs}\label{outputs-10-005}
 
 This output is the operating part-load ratio of the boiler. The part-load ratio is calculated as the boiler load divided by the boiler nominal capacity. The part-load ratio is limited by the minimum and maximum part-load ratio inputs specified by the user.
 
+\paragraph{Boiler Efficiency {[]}}\label{boiler-efficiency}
+
```
Comment (mitchute) [inline doc/input-output-reference/src/overview/group-plant-equipment.tex:4165]: I like the note, but just for the sake of uniformity and completeness, we should probably also have an explicit definition of efficiency. Maybe add something like "this output is boiler heating rate divided by boiler gas rate" or similar between the label and the note.

File: doc/input-output-reference/src/overview/group-plant-equipment.tex [hunk 2]
```diff
@@ -4401,6 +4409,10 @@ \subsubsection{Outputs}\label{outputs-11-004}
 
 These outputs are the heating output (load) of the boiler. Energy is metered on Boilers:EnergyTransfer, EnergyTransfer:Plant, and EnergyTransfer:Facility.
 
+\paragraph{Boiler Steam Efficiency {[]}}\label{boiler-steam-efficiency}
+
```
Comment (mitchute) [inline doc/input-output-reference/src/overview/group-plant-equipment.tex:4415]: Same change here.

File: src/EnergyPlus/BoilerSteam.cc [hunk 3]
```diff
@@ -214,7 +214,7 @@ namespace BoilerSteam {
                 ShowWarningMessage(DataIPShortCuts::cCurrentModuleObject + "=\"" + DataIPShortCuts::cAlphaArgs(1) + "\"");
                 ShowContinueError("Field: Maximum Operation Pressure units are Pa. Verify units.");
             }
-            thisBoiler.Effic = DataIPShortCuts::rNumericArgs(2);
+            thisBoiler.NomEffic = DataIPShortCuts::rNumericArgs(2);
```
Comment (mitchute) [inline src/EnergyPlus/BoilerSteam.cc:217]: More explicit. This is good.

File: tst/EnergyPlus/unit/BoilerHotWater.unit.cc [hunk 4]
```diff
@@ -185,3 +191,95 @@ TEST_F(EnergyPlusFixture, Boiler_HotWater_BlankDesignWaterFlowRate)
     // Additional tests for fuel type input
     EXPECT_EQ(state.dataBoilers.Boiler(1).BoilerFuelTypeForOutputVariable, "Gas");
 }
+
```
Comment (mitchute) [inline tst/EnergyPlus/unit/BoilerHotWater.unit.cc:194]: Unit tests look good.

**Review by mitchute — CHANGES_REQUESTED** [review]
Minor doc changes requested in the IORef but this looks good other than that. Running tests locally now.

**Comment by mitchute** [comment]
Unit tests look good, and I ran a few files and the results look good. We also need to add the new output variables to a least one file for the Boiler:HotWater and Boiler:Steam objects.

**Comment by matthew-larson** [comment]
@mitchute I added descriptions for the output variables as requested and also added the output variables to a few files with Boiler:HotWater and Boiler:Steam objects that were outputting similar variables.

**Review by mitchute — APPROVED** [review]

**Comment by mitchute** [comment]
Thanks, @matthew-larson. This should go in once CI reports back.

**Comment by matthew-larson** [comment]
Just pushed up changes to fix the build errors.

**Comment by mitchute** [comment]
AUD and RDD diffs are expected here. Merging.

**Pull request merged**